Added DNPtime support

SElyPer opened the pull request.

SElyPer committed:
DNPTime patch

SElyPer committed:
Adding all files

jadamcrain commented:
Hi,

We'll need to get this PR down to just the necessary file changes.

There are a lot of non-relevant changes in here. As an example, the Java C++ code has changed locations from /java/cpp/jni to cpp/jni. I'm open to discussion about changing the locations of things, but need individual issues to be addressed in individual PRs.

Try rebasing just the changes for the g50v1 time support. It would even be best to do individual commits, e.g. add support to C++, then to Java, etc.

-Adam

SElyPer commented:
Hey Adam,
I'm having a hell of a time paring it down with all of the automated
generation.  Without changing any files, if I check out your development
branch, build the codegen and generation projects, I get errors due to the
java/api directory missing.  At this point Git shows well over a hundred
files that have been modified; mostly due to the Copyright header being
added.  Several other files report being modified because the code is in
slightly different order (see the image below).  If you can clean up the
auto generation to not create phantom changes, I can easily provide the
changed files.  What happened to the API directory?

[image: Inline image 1]

Attached are the SCALA files which produce the generation changes along
with the modified java and cpp files that I believe make up the changes I
created.  Unfortunately, I can't easily identify the files which are
different given the copyright and code reordering issues mentioned above.

Thoughts?

-Sam

On Tue, Oct 25, 2016 at 11:08 AM, Adam Crain notifications@github.com
wrote:

> Hi,
> 
> We'll need to get this PR down to just the necessary file changes.
> 
> There are a lot of non-relevant changes in here. As an example, the Java
> C++ code has changed locations from /java/cpp/jni to cpp/jni. I'm open to
> discussion about changing the locations of things, but need individual
> issues to be addressed in individual PRs.
> 
> Try rebasing just the changes for the g50v1 time support. It would even be
> best to do individual commits, e.g. add support to C++, then to Java, etc.
> 
> -Adam
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/automatak/dnp3/pull/183#issuecomment-256063019, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AV-VRPF0URhRwi--gp9OGH1rFOxTYMyZks5q3htrgaJpZM4KgEmH
> .

## 

Sam Ely
President, Percival Inc.
SDVOSB
443-420-7002

jadamcrain commented:
Can you email me diffs of the important changes, and I'll try to put this in today.

Part of the problem is that the development branch is still under heavy active development and things have been changing underneath you.

Regarding the API directory, I merged the bindings/api JARs  b/c there was no good reason to treat them separately.

If you can just give me diffs of the important changes, I think I can just put this functionality in here for you.

-Adam

jadamcrain commented:
Closing PR since the same functionality was added in another commit.
